src/header.rs

```diff
@@ -268,16 +273,11 @@ impl Header {
         if self.v3.refcount_order > 6 {
             return Err(Error::FileFormat(format!("bad refcount_order {}", self.v3.refcount_order)));
         }
-        if self.v3.header_length as u64 != io.position() {
+        if self.v3.header_length as u64 != actual_length {
```

Comment:
I'm not sure if this check is actually useful. We're setting position to header_length, storing that, and then checking if it's what we set it to. Of course it is!

Maybe instead we can check if the length matches the spec? https://github.com/qemu/qemu/blob/master/docs/interop/qcow2.txt#L184-L185